### PR TITLE
Shorten description on search page

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

The description on the search page was quite wordy. This PR reduces clutter and reproduces my accepted PR at https://github.com/sphinx-doc/sphinx/pull/6814/.

### CODE REVIEW

File: doc/_templates/search.html [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
-    search function will automatically search for all of the
-    words. Pages containing less words won't appear in the result
-    list.{% endtrans %}  If you want to limit your search to working code examples,
-    include the keyword "codex" (mnemonic for <i>code example</i>) in your
-    search, e.g., "codex ellipse".
+    {% trans %}Searching for multiple words only shows matches that contain
+    all words.{% endtrans %}
+  </p>
+  <p>
+    If you want to limit your search to working code examples, including the
```
Comment (dstansby) [inline doc/_templates/search.html]:
```suggestion
    If you want to limit your search to working code examples, include the
```

File: doc/_templates/search.html [hunk 2]
```diff
@@ -7,13 +7,13 @@
...
   <h1 id="search-documentation">{{ _('Search') }}</h1>
   <p>
-    {% trans %}From here you can search these documents. Enter your
-    search words into the box below and click "search". Note that the
-    search function will automatically search for all of the
-    words. Pages containing less words won't appear in the result
-    list.{% endtrans %}  If you want to limit your search to working code examples,
-    include the keyword "codex" (mnemonic for <i>code example</i>) in your
-    search, e.g., "codex ellipse".
+    {% trans %}Searching for multiple words only shows matches that contain
```
Comment (story645) [inline doc/_templates/search.html]: Is this true even when comma delimited?

Comment (timhoffm) [reply doc/_templates/search.html]: Yes, "StarPolygonCollection Quiver" and "StarPolygonCollection, Quiver" yield the same two entries.

**Comment by timhoffm** [comment]
I was hoping that this backports cleanly. Since it does not, it's not worth the hassle of backporting manually.

**Pull request merged**